samcli/lib/runtimes/base.py

```diff
@@ -0,0 +1,342 @@
+        Family.NODEJS,
+        "amazon/nodejs20.x-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    nodejs18x = (
+        "nodejs18.x",
+        Family.NODEJS,
+        "amazon/nodejs18.x-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    nodejs16x = (
+        "nodejs16.x",
+        Family.NODEJS,
+        "amazon/nodejs16.x-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+
+    provided_al2023 = (
+        "provided.al2023",
+        Family.PROVIDED,
+        None,
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    provided_al2 = (
+        "provided.al2",
+        Family.PROVIDED,
+        None,
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    provided = (
+        "provided",
+        Family.PROVIDED,
+        None,
+        [Architecture.X86_64],
+    )
+    go_provided_al2023 = (
+        "go (provided.al2023)",
+        Family.PROVIDED,
+        "amazon/go-provided.al2023-base",
+        [Architecture.X86_64, Architecture.ARM64],
+        False,
+    )
+    go_provided_al2 = (
+        "go (provided.al2)",
+        Family.PROVIDED,
+        "amazon/go-provided.al2-base",
+        [Architecture.X86_64, Architecture.ARM64],
+        False,
+    )
+
+    python312 = (
+        "python3.12",
+        Family.PYTHON,
+        "amazon/python3.12-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    python311 = (
+        "python3.11",
+        Family.PYTHON,
+        "amazon/python3.11-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    python310 = (
+        "python3.10",
+        Family.PYTHON,
+        "amazon/python3.10-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    python39 = (
+        "python3.9",
+        Family.PYTHON,
+        "amazon/python3.9-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    python38 = (
+        "python3.8",
+        Family.PYTHON,
+        "amazon/python3.8-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+
+    ruby32 = (
+        "ruby3.2",
+        Family.RUBY,
+        "amazon/ruby3.2-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+
+
+@unique
+class DeprecatedRuntime(RuntimeEnumBase):
+    dotnetcore20 = ("dotnetcore2.0", Family.DOTNET)
+    dotnetcore21 = ("dotnetcore2.1", Family.DOTNET)
+    dotnetcore31 = ("dotnetcore3.1", Family.DOTNET)
+    nodejs43 = ("nodejs4.3", Family.NODEJS)
+    nodejs610 = ("nodejs6.10", Family.NODEJS)
+    nodejs810 = ("nodejs8.10", Family.NODEJS)
+    nodejs10x = ("nodejs10.x", Family.NODEJS)
+    nodejs12x = ("nodejs12.x", Family.NODEJS)
+    nodejs14x = ("nodejs14.x", Family.NODEJS)
+    python27 = ("python2.7", Family.PYTHON)
+    python36 = ("python3.6", Family.PYTHON)
+    python37 = ("python3.7", Family.PYTHON)
+    ruby25 = ("ruby2.5", Family.RUBY)
+    ruby27 = ("ruby2.7", Family.RUBY)
+
+
+def runtime_dep_template_mapping(runtimes: List[RuntimeDataMixin]) -> dict:
```

Comment:
nit: can we use full names in the method, I am not sure if `dep` is for dependency or something else?